reactioncommerce/reaction
# [Marketplace] Feature - Permissions should be settable by placing a user into a predefined group

Permission groups. Implements #2184 
This PR has the Meteor methods to manage permission groups. It specifically adds `group/createGroup`, `group/updateGroup`, `group/addUser`, `group/removeUser`.
The UI for this is being handled separately (as that involved changing a lot of the current Accounts section of admin dashboard).

**To test:**
_ Take a look at the tests, to have a sense of how the methods work
_ The methods can be tested from the browser console of a running Reaction (via `Meteor.call` and passing the needed values). See the jsdocs added for the descriptions of the arguments.
* Start Reaction
* Create a shop (i.e register as a user (not the owner). Go to "Profile", click "Become a Seller"
* Create a group, by calling `group/createGroup` from console, passing in the required options.
* Confirm in db that a new doc is created in Groups collection
* Add a user to the group by calling `group/addUser` as described
* Confirm that the user's permissions gets updated (i.e Meteor.user.roles) and that the user's Account doc now has the groupId.
* The other methods follow similar pattern as shown in the tests

Other notes:
_ Groups exits as a schema
_ Group Ids are referenced on Accounts to track which users belong to a group
